WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m

```diff
@@ -606,48 +618,123 @@ - (UITableViewCell *)configureFeaturedImageCellForIndexPath:(NSIndexPath *)index
     return cell;
 }
 
-- (UITableViewCell *)configureGeolocationCellForIndexPath:(NSIndexPath *)indexPath
-{
-    WPTableViewCell *cell;
-    if (self.post.geolocation == nil) {
-        WPTableViewActivityCell *actCell = [self getWPActivityTableViewCell];
-
-        actCell.tag = PostSettingsRowGeolocationAdd;
-
-        if ([[LocationService sharedService] locationServiceRunning]) {
-            [actCell.spinner startAnimating];
-            actCell.textLabel.text = NSLocalizedString(@"Finding your location...", @"Geo-tagging posts, status message when geolocation is found.");
-        } else {
-            actCell.textLabel.text = NSLocalizedString(@"Set Location", @"Geolocation feature to set the location.");
-            [actCell.spinner stopAnimating];
-        }
-
-        cell = actCell;
-
+- (void)showLocationError:(NSError *)error {
+    NSString *title = NSLocalizedString(@"Location", @"Title for alert when a generic error happened when trying to find the location of the device");
+    NSString *message = NSLocalizedString(@"There was a problem when trying to access your location. Please try again later.",  @"Explaining to the user there was an error trying to obtain the current location of the user.");
+    NSString *cancelText = NSLocalizedString(@"OK", "");
+    NSString *otherButtonTitle = nil;
+    if (error.domain == kCLErrorDomain && error.code == kCLErrorDenied) {
+        otherButtonTitle = NSLocalizedString(@"Open Settings", @"Go to the settings app");
+        title = NSLocalizedString(@"Location", @"Title for alert when access to the media library is not granted by the user");
+        message = NSLocalizedString(@"WordPress needs permission to access your device location in order to geotag your post. Please change the privacy settings if you wish to allow this.",  @"Explaining to the user why the app needs access to the device location.");
```

Comment:
I think we're not saying "geotag" anywhere else, so it felt a bit weird to see here.  Maybe we could wordsmith this message some?  Another take on it might be `WordPress needs permission to add your location to the post. Please update your privacy settings.` but that feels a bit terse. Maybe we can get some editorial input?